testing/e2e/manifests/clientServer.go

```diff
@@ -0,0 +1,145 @@
+package manifests
+
+import (
+	_ "embed"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	netv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+//go:embed embedded/client.go
+var clientContents string
+
+//go:embed embedded/server.go
+var serverContents string
+
+type testingResources struct {
+	Client  *appsv1.Deployment
+	Server  *appsv1.Deployment
+	Service *corev1.Service
+	Ingress *netv1.Ingress
+}
+
+func (t testingResources) Objects() []client.Object {
+	ret := []client.Object{
+		t.Client,
+		t.Server,
+		t.Service,
+		t.Ingress,
+	}
+
+	for _, obj := range ret {
+		setGroupKindVersion(obj)
+	}
+
+	return ret
+}
+
+func ClientAndServer(namespace, name, zoneName, nameserver, keyvaultURI string) testingResources {
+	host := strings.ToLower(namespace) + "." + zoneName
+
+	clientDeployment := newGoDeployment(clientContents, namespace, name+"-client")
```

Comment:
We cannot. We shouldn't mix things here and in RP because this is OSS. Also this repo is the source of any new code changes so this repo would be the source of truth (not RP if we wanted to do things that way).

Reply:
that's a good point - wonder if it's worth putting client/server functionality in an exportable module and having RP import that